Implement Accordion Block Plugin

jffng opened the pull request:
**What**
This PR implements an Accordion block. It's an alternative to #6.

![Screenshot 2024-07-19 at 4 03 59 PM](https://github.com/user-attachments/assets/a8ee6d9a-e24f-4cd5-acb6-1f0ac579bf7b)

This approach actually is composed using four blocks:

- Accordion
   - Accordion Item
     - Accordion Trigger
     - Accordion Content

Without getting too much into the weeds, this was done because it was the simplest way I could figure out how to allow user styling of the trigger and content separately, while maintaining the accessibility and semantic needs of an accordion pattern.

**Highlights**
- Allows you to configure which items are open by default, and the expand / collapse behavior
- Comes with a few block styles for the icons
- Informed by design mockups by @jarekmorawski
- Uses the Interactivity API
- Meets WC3 accessibility guidelines

**Demo & Download**
https://cool-tools.mystagingwebsite.com/2024/07/12/accordion/

**Video Walkthrough**
https://video.wordpress.com/embed/AfzRE5Om

jffng committed:
Initial commit.

jffng committed:
Use interactivity API.

jffng committed:
Incorporate heading level.

jffng committed:
Add text align.

jffng committed:
Scaffold an icon for the accordion title.

jffng committed:
Convert to HTML tag processor.

jffng committed:
Make classnames consistent.

jffng committed:
Fix accordion within accordion bug.

jffng committed:
Add toggle and format.

jffng committed:
allow text align to work.

jffng committed:
Add layout support

jffng committed:
Downgrade wp-scripts for now.

jffng committed:
Add icon styles.

jffng committed:
Use background as mask instead.

jffng committed:
Rename to accordion-item.

jffng committed:
Format.

jffng committed:
Format.

jffng committed:
Update colors.

jffng committed:
Add setting for icon position.

jffng committed:
Allow no icon option.

jffng committed:
Replace with custom icon.

jffng committed:
Convert to two blocks.

jffng committed:
Refactor iAPI.

jffng committed:
Fix icon display.

jffng committed:
Remove unused CSS for now.

jffng committed:
Make id more specific.

jffng committed:
Remove zip.

jffng committed:
Remove comments.

jffng committed:
Allow multiple accordion items.

jffng committed:
WIP.

jffng committed:
Fix array filtering bug.

jffng committed:
Cleanup unused classes.

jffng committed:
Refactor so styles live at top level block and no need for textAlign.

Remove unnecessary build files.

jffng committed:
Update server side register with correct block names.

jffng committed:
Simplify to accordion item.

jffng committed:
Fix classes and format.

jffng committed:
Add support for border and typography.

jffng committed:
Enable more typography and border support.

jffng committed:
Remove build files.

jffng committed:
Ignore package-lock

jffng committed:
Remove editorconfig.

jffng committed:
Adds css grid based animation.

jffng committed:
Format.

jffng committed:
Bump wordpress/scripts to highest 27.

jffng committed:
Add color support.

jffng committed:
Enable multiple items to be open at once.

jffng committed:
Separate trigger and content blocks.

jffng committed:
Enable template locking.

jffng committed:
Enable open by default.

jffng committed:
Clean up.

jffng committed:
Replace icons with confusing ones.

jffng committed:
Enable better icon positioning and sizing.

jffng committed:
Support anchors.

jffng committed:
Fix aria implementation.

jffng committed:
Bump version.

jffng committed:
Fix how styles are applied.

jffng commented:
@jarekmorawski @Nyiriland would you be able to give this a review when you have a chance? 🙏

jarekmorawski commented:
Just took it for a spin. Great work, @jffng! I assure you it'll make a big difference for 1000s of Woo stores out there. It's much more capable and easier to use than core's Details block. I also like the animation. It brings the block on par with other existing solutions out there.

> Without getting too much into the weeds, this was done because it was the simplest way I could figure out how to allow user styling of the trigger and controls separately, while maintaining the accessibility and semantic needs of an accordion pattern.

I wonder if we should call the block `Accordion Group` or `Accordions`, then? And `Accordion Item` to just `Accordion`? And add a small inline `+` in the bottom right corner that would let users easily add new items? It seems the block works similarly to the Buttons block so the more UX we can re-use, the better for the user.

Clicking the black + would immediately insert a new Accordion block.

<img width="388" alt="image" src="https://github.com/user-attachments/assets/92e26e3a-2b0a-46ec-9475-2053a7ba3add">

Other thoughts:

<img width="305" alt="image" src="https://github.com/user-attachments/assets/9ec8bd35-08dc-40ef-832d-be6014e0d5ef">

- I'd all this section must `Settings`.
- The icon position button should probably be full-width to match other similar controls in the UI.
- Can we move all icon-related settings to a new section in the settings tab? It'd be nice to use the actual icons instead of words (unless icons are turned off in the editor's accessibility settings).

<img width="524" alt="image" src="https://github.com/user-attachments/assets/6285adc9-0ddb-4749-9f03-0402dfbd13be">

- Then under Styles, users could decide if they want to keep the icon or not (`Default` would mean the icon is on). When "Text only" is active, we wouldn't show the `Icon` section in Settings.

<img width="492" alt="image" src="https://github.com/user-attachments/assets/99b9b0f6-5744-4f36-a8b2-355319673c00">

- (Nice to have) It'd be amazing to expose all layout-related settings in the Layout section, including `Width` and alignment. I think the former is only compatible with group blocks right now, but we should have some width settings. Users may not wish for the accordion to fill the entire available space.

<img width="235" alt="image" src="https://github.com/user-attachments/assets/7f9a4195-3013-4b91-8162-0de13be76092">

As for the inner blocks, I'd rename `Accordion Trigger` to `Trigger` and `Accordion Container` to just `Content` and change the icons to some that are visually related and share a common metaphor (the arrow). [See icons in Figma](https://www.figma.com/design/mXdR6fkq8XqyWBFxvpyDOn/%5BProject%5D-PDP-Blocks?node-id=25-299&t=hqKWpE4Ld0nFNA5U-11)

<img width="757" alt="image" src="https://github.com/user-attachments/assets/f929c0f8-9f48-437e-aee8-bd1963b6c7fc">

That's all I had. Again, _great_ work. Thanks for doing this!

Nyiriland commented:
> Then under Styles, users could decide if they want to keep the icon or not (Default would mean the icon is on). When "Text only" is active, we wouldn't show the Icon section in Settings.

Instead of introducing style options, what if we add a "None" option for the icon? The benefit is that this keeps related options together, and if developers want to extend the block by adding new styles, they won't compete with a "no icon" option (since you can only add one block style via the current Styles UI).

<img width="512" alt="image" src="https://github.com/user-attachments/assets/7c2d6844-22fa-48b9-90f8-59ef00401e9c">

- Should the accordion be closed, unless you select the block, or have "open by default" checked? Similar to how the Details block works. This would ensure a closer editor/front-end experience.

- Do the "accordion trigger" and "accordion content" need to be separate item in the block list view? From [reading Jeff's initial comment](https://github.com/a8cteam51/special-projects-blocks-monorepo/pull/16#issue-2419761777), perhaps they do. My concern is that it might feel over-engineered for amateur users compared to the details block, but maybe that's needed for a more comprehensive block system:

<img width="545" alt="image" src="https://github.com/user-attachments/assets/c370840b-9523-4e8a-a1aa-48aaa71a45e2">

- In the most basic example of how I'd want to style, I'd want to add a border to the top & bottom of the Accordion Item (see mockup below).

![image](https://github.com/user-attachments/assets/cb0623b6-21aa-4cf3-927f-ab0ea46a8a44)

![image](https://github.com/user-attachments/assets/099ce707-c95a-4759-82b4-034053bc3f3c)

From testing, you can add the border to Accordion Item, but not padding. If you add padding to the bottom of the Accordion Content, it does this:

https://github.com/user-attachments/assets/e8c30c6b-2da1-4e8c-9885-9129e4c4a43c

- Icons: are there default icons in the WordPress UI library that we can utilize instead of adding new ones? I know we have a WP icon library in Figma, but not sure if you can access the UI icons when developing to include in CSS.

- Suggested icon animation details:
<img width="599" alt="image" src="https://github.com/user-attachments/assets/462fb95e-f831-4537-9c1b-8fb2a623c1db">

jarekmorawski commented:
> Instead of introducing style options, what if we add a "None" option for the icon?

Yeah, that's another possibility. There's no consensus on how such binary styles (icon vs. no icon) should be handled on the block side so we have some freedom.

> Should the accordion be closed, unless you select the block, or have "open by default" checked?

Agreed!

> My concern is that it might feel over-engineered for amateur users compared to the details block, but maybe that's needed for a more comprehensive block system:

That's my concern too but I had a chat with Jeff and it seems there's no other solution that wouldn't break a11y and diverge from HTML semantics. In the original design, the structure was much simpler.

<img width="387" alt="image" src="https://github.com/user-attachments/assets/df75d6ab-a994-4e88-9fe5-323fced392a4">

Perhaps it's worth a second look? I can ask Woo block engineers to share their thoughts.

> From testing, you can add the border to Accordion Item, but not padding. If you add padding to the bottom of the Accordion Content, it does this:

It could be a bug. I don't think we should reveal the content when the accordion is not open.

@Nyiriland, I was able to recreate that layout using a combination of Accordion Trigger padding and block spacing added to both Accordion Item and Accordion (the parent block). It's a bit convoluted, but well, designing complex nested layouts in Figma isn't a walk in the park either. 😉 

<img width="803" alt="image" src="https://github.com/user-attachments/assets/59a64c21-b335-4111-8ec7-7d6dcf511790">

> Icons: are there default icons in the WordPress UI library that we can utilize instead of adding new ones?

We can, but they may confuse users who may be familiar with blocks that already use those icons. We use custom icons all the time at Woo and it doesn't seem to be an issue. If we were to go with the default icons, I'd suggest:

- Accordions: `list`
- Accordion: `details`
- Trigger: `queryPaginationNext`
- Content: `termDescription`

<img width="604" alt="image" src="https://github.com/user-attachments/assets/998e239a-8278-4ca5-84f6-35c1ad1d30b2">

jffng commented:
Thank you @jarekmorawski @Nyiriland, great feedback!

> I wonder if we should call the block `Accordion Group` or `Accordions`, then? And `Accordion Item` to just `Accordion`?

The way I understand the UX pattern and its naming, is that a single Accordion contains several sections of content that can be expanded and collapsed. The visual metaphor is of an instrument that expands in one place and contracts in another. So technically, there should be multiple sections (or "items") to make it an accordion, otherwise there the distinction between an `Accordion` and `Details` would be even more blurry. Does that make sense? Sorry if it's a bit esoteric. 

> As for the inner blocks, I'd rename `Accordion Trigger` to `Trigger` and `Accordion Container` to just `Content` and change the icons to some that are visually related and share a common metaphor (the arrow).

👍 I agree, sounds good.

> Do the "accordion trigger" and "accordion content" need to be separate item in the block list view? 

> Perhaps it's worth a second look? I can ask Woo block engineers to share their thoughts.

I'm open to ideas. I hit a wall in this PR https://github.com/a8cteam51/special-projects-blocks-monorepo/pull/6 which tries to do that but fell short of the requirements, particularly the ability to easily style the trigger, separately from the content, using the existing block supports and maintaining semantics.

Something to keep in mind with the design of the block's editing experience: there needs to be a way to associate a trigger with its corresponding content. This is what the `Item` does — it just associates a trigger with content and disallows either from being moved or removed, and any other block types from being inserted, since doing so would break the functionality and UX pattern. 

This approach also allows the user to insert as many Items into the Accordion as needed. I couldn't think of another way to keep the trigger and content in the right order. (AFAIK there is not a simple way in the block editor API to say, "only allow inserting a content block after a trigger")

> From testing, you can add the border to Accordion Item, but not padding. If you add padding to the bottom of the Accordion Content, it does this:

> It could be a bug. I don't think we should reveal the content when the accordion is not open.

Yeah, that's a bug. It's related to using a CSS-only approach to animating the show / hide behavior. With this approach, no hacky JS height calculations are needed, and it's responsive, which is great. But the result is that bug happens when padding is added to the `Content` container block. To get around it, I would need to disallow setting top and bottom padding on the `Content` block. This way, to add vertical padding to the Content, the user would have to wrap the blocks inside `Content` in a group block.  It's convoluted, but I think a worthwhile tradeoff. What do you think? 

> Icons: are there default icons in the WordPress UI library that we can utilize instead of adding new ones?

I think we should supply custom icons, the core ones should be left specific to their respective blocks and settings.

I'll get started on the rest of the feedback and let you know when it's ready for another test.

Nyiriland commented:
> We can, but they may confuse users who may be familiar with blocks that already use those icons. 

Sorry, I was referring to the icons for the trigger, not the block itself. The [Twentig plugin](https://twentig.com/wordpress-blocks/) extends the Details block to create accordion options, and I was noticing that the trigger icons look a lot like the WP icon set. If they're using the same ones, we probably can, too. (We chose not to extend the Details block because the markup was not built in an accessible way.)

<img width="486" alt="image" src="https://github.com/user-attachments/assets/ae99ab96-cd75-4839-abc9-b567405aaa9c">

But if not, the icons @jarekmorawski used in his mockups suffice.

jffng commented:
@jarekmorawski for some reason when I try to export the `Content` block icon from Figma and load it into the block, I'm getting this: 

<img width="350" alt="image" src="https://github.com/user-attachments/assets/4b229236-66f8-44ae-a88d-ed2a823bc1a8">

The other block icons look alright, do you know of a way that might export the icon correctly?

jffng committed:
Replace icons and simplify names.

jffng committed:
Remove padding controls because it messes up the show/hide behavior.

jffng committed:
Change the icon open/close states.

jffng committed:
Fix icon color inheritance.

jffng committed:
Add transition for only users who have no preference set.

Nyiriland commented:
> Yeah, that's a bug. It's related to using a CSS-only approach to animating the show / hide behavior. With this approach, no hacky JS height calculations are needed, and it's responsive, which is great. But the result is that bug happens when padding is added to the Content container block. 

I wouldn't even call it a bug, it's how padding works. 😆 I'm more concerned this is how people might expect to apply padding/spacing to block content, so we should account for that.

> To get around it, I would need to disallow setting top and bottom padding on the Content block. This way, to add vertical padding to the Content, the user would have to wrap the blocks inside Content in a group block. It's convoluted, but I think a worthwhile tradeoff. What do you think?

I agree that a CSS-only solution for expand/collapse is cool, but I don't love having to wrap content in a group block just to be able to add spacing around it; I don't think it's an intuitive solution for beginner users that just want to be able to throw in content and expect it to look "ok". In my opinion, that includes having enough spacing between the accordion content and the block below it.

Idea: is it possible to add a simple `div` around the Content that shows/hides it? That way, we can apply padding/spacing to the Content, but it wouldn't be visible in the closed state.

Nyiriland commented:
<img width="251" alt="image" src="https://github.com/user-attachments/assets/174deace-e0a4-4fd6-b549-6dddeb82eec1">

Oh one more thing: for this option I'd **default** the block to allowing multiple items open. It's better for UX in my opinion, and [Nielson Norman](https://www.nngroup.com/articles/accordions-on-desktop/) agrees:

> This problem can further escalate when an accordion automatically closes when another is opened, restricting users’ ability to combine information from multiple accordions at the same time. If you expect users to need information from several accordions at once, it is better to display all the content at once (even if it results in a longer page).

jarekmorawski commented:
> So technically, there should be multiple sections (or "items")

In such a case, do we need the Content block? Couldn't everything inside the accordion block that _isn't_ the trigger be treated as content? That's that tripped me off; if Content is the container, why couldn't I put inside multiple sections inside like you describe?

> The [Twentig plugin](https://twentig.com/wordpress-blocks/) extends the Details block to create accordion options, and I was noticing that the trigger icons look a lot like the WP icon set

Those icons don't come from the default WP set, but they do look similar. We can create our own and make sure they match what's being displayed on the front-end. AFAIK, there's no easy way to display individual Gutenberg icons on the front-end, so we may need to use custom icons anyway.

Added a bunch to the [Figma](https://www.figma.com/design/mXdR6fkq8XqyWBFxvpyDOn/%5BProject%5D-PDP-Blocks?node-id=1108-25518&t=VfsMfqrq0JZT4ZWE-11).

<img width="543" alt="image" src="https://github.com/user-attachments/assets/0f9b5859-6d35-4608-866f-6ca9fcb93816">

> It's better for UX in my opinion

100% agree.

@jffng, one more thing: can we repeat some settings across the inner blocks and keep them in sync? For example, it'd be great to show the icon picker when the trigger block is selected. Currently, it's only displayed in the parent, which some folks may never discover. The same goes for the icon position.

> for some reason when I try to export the Content block icon from Figma and load it into the block, I'm getting this:

Here's the SVG file! Let me know if it works.

![content](https://github.com/user-attachments/assets/face8709-c5f0-411b-9d4d-71e814fbe7b8)

jffng commented:
> Here's the SVG file! Let me know if it works.

Hmm that link is not allowing me to download it.

Nyiriland commented:
> Hmm that link is not allowing me to download it.

@jarekmorawski Can you "copy as svg" from Figma and paste in a code block (or point me to the icon in Figma)? There might be some issues with masking or layer order in the SVG code, I can help sort it out if needed.

jffng committed:
Fix clip and fillRule.

jffng committed:
Use wrapping div so padding can be applied as expected.

jffng committed:
Change setting name to autoclose and make false by default.

jffng committed:
Fix icon.

jffng committed:
Move icons to settings of trigger.

jarekmorawski commented:
Sure thing. FWIW, you can right-click on that SVG and download it like any other image. Here's the code:

`<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<path fill-rule="evenodd" clip-rule="evenodd" d="M8.10417 6.00024H6.5C5.39543 6.00024 4.5 6.89567 4.5 8.00024V10.3336H6V8.00024C6 7.7241 6.22386 7.50024 6.5 7.50024H8.10417V6.00024ZM4.5 13.6669V16.0002C4.5 17.1048 5.39543 18.0002 6.5 18.0002H8.10417V16.5002H6.5C6.22386 16.5002 6 16.2764 6 16.0002V13.6669H4.5ZM10.3958 6.00024V7.50024H13.6042V6.00024H10.3958ZM15.8958 6.00024V7.50024H17.5C17.7761 7.50024 18 7.7241 18 8.00024V10.3336H19.5V8.00024C19.5 6.89567 18.6046 6.00024 17.5 6.00024H15.8958ZM19.5 13.6669H18V16.0002C18 16.2764 17.7761 16.5002 17.5 16.5002H15.8958V18.0002H17.5C18.6046 18.0002 19.5 17.1048 19.5 16.0002V13.6669ZM13.6042 18.0002V16.5002H10.3958V18.0002H13.6042Z" fill="#1E1E1E"/>
</svg>`

jffng commented:
Assuming the block had to stay with this structure structure, what do you think about this naming?

<img width="543" alt="image" src="https://github.com/user-attachments/assets/76b16102-5151-4e98-b6ad-c214079d2011">

jffng committed:
Rename.

jffng committed:
Reorder and label the icon options.

jffng committed:
Allow background image and only vertical margin.

jffng committed:
Improve insertion UX.

jffng committed:
Add back hidden html attribute.

jffng committed:
WIP determine if block is selected.

jffng committed:
Auto expand and close based on selection.

jffng committed:
More sensible placeholders.

jffng committed:
Fix ariaHidden -> aria-hidden.

jffng committed:
Bump minor version.

Bump version.

jffng committed:
Force margin on wrapper div.

jffng commented:
Okay, I made the following changes: 

- Matched the editing experience to the front-end so that accordion content is only visible if the block is selected or it's set to be open by default
- Fixed the padding issue on the `Content`, shoutout @Nyiriland!! 
- Updated the UI icons, so they are now in the block settings of the `Trigger`. I'm not sure about repeating settings / keeping them in sync across blocks, is there an example of this you have seen? Happy to explore in a follow up.
- Renamed the blocks to `Accordion Group` > `Accordion Item` > `Trigger` + `Content`
- Updated the block icons
- Renamed the "Allow multiple items open" setting to "Autoclose", and it's off by default

A demo and plugin download is available at https://cool-tools.mystagingwebsite.com/2024/07/12/accordion/

https://cloudup.com/cBIxzqcM2Ym

One follow up for sure I want to do is to add a setting to adjust the icon size.

Nyiriland commented:
Wow, looking awesome!!! Love all the options for the trigger 🤡

- Changing styles: difficult to change on multiple accordions; you either have to copy styles, or duplicate one you've already styled
- Let's look into if it's possible to register blocks in "global styles" to make this ^ easier
- Can we add ability a border to the `accordion group`? Specifically, if you have a bottom border on all your accordions, you should be able to add a border to the top of the **group**: this will allow you to re-order the child elements, while still ensuring there's always a top border.
- I know this may be a theme-level style change, but changing the `focus` style on the trigger to `focus-visible` would make that irritating blue border disappear when you click it.
- Could we have an option to **turn off** the open/close animation transition on the group? (But keep it on by default)
- I think the "Autoclose" label needs some rewording so it's more clear what it does (or at least change to "Auto-close"... I'll think on this)

I'm thinking about any "default styles" we might want to include (@jarekmorawski may have input here). To me, this is pretty much the default I'd want to go for:

<img width="711" alt="image" src="https://github.com/user-attachments/assets/522b6be2-e601-473f-b184-3c98cdabe61f">

To achieve this: 

- Zero block spacing on `accordion group`
- 1px border-top on `accordion group`
- 1px border-bottom on `accordions`
- `spacing 2` on top & bottom of `accordion trigger`
- `spacing 2` on bottom of `content`

jarekmorawski commented:
> Changing styles: difficult to change on multiple accordions; you either have to copy styles, or duplicate one you've already styled

Global Styles is one option, multi-select is another. In the editor, you can select a few blocks of the same type to see and modify their settings. I'm not sure if that requires an extra API or something, but it's worth looking into.

> To me, this is pretty much the default I'd want to go for:

100% agree. It'd be great to have a default style like this. In the future, we can add a few more as block styles.

On top of that, I have a few extra comments:

I don't think the empty option is working in the icon picker. It took me a second to understand what it was. We can either replace it with `None` (text button) or add a toggle `Display icon` that will display the picker when enabled.

<img width="309" alt="image" src="https://github.com/user-attachments/assets/1ec0ae7a-acc5-414d-b5f2-f7dc7ac72522">

It'd be nice if the newly added accordion block would replicate the recently used styles. Otherwise, we force the user to style it from scratch or resort to duplicating existing accordions.

<img width="996" alt="image" src="https://github.com/user-attachments/assets/2f91c179-8f3d-4437-87d7-9674246be4e4">

> I think the "Autoclose" label needs some rewording so it's more clear what it does (or at least change to "Auto-close"... I'll think on this)

I'd rename it to `Close automatically` and add a help text: `Automatically close this accordion when visitors open another one`.

H3 feels too big for a default style. Can we try H5?

<img width="512" alt="image" src="https://github.com/user-attachments/assets/d1201fa9-c904-4c85-b8a5-857bc9e4ac23">

If we are to add a toggle to disable animations, I'd put it `Advanced` so it doesn't get in most people's way.

Nyiriland commented:
Good points on everything, @jarekmorawski. 

> We can either replace it with None (text button) or add a toggle Display icon that will display the picker when enabled.

I also didn't notice that there was a "none" option for the icon, hah. I'd vote for `None` as text, or remove the option altogether. (For a11y, there _should_ be an indicator to imply additional functionality somewhere.)

> H3 feels too big for a default style. Can we try H5?

I tested this out with 2024, and h5 feels too small ([I'm stress testing over on this page](https://cool-tools.mystagingwebsite.com/test/), btw). h4 works better (although I am also still a fan of h3, too). @jffng  is there any guidance around semantic hierarchy of DOM elements for block defaults? Like, I'd almost want to make it an h2, then size the text smaller, but that's probably a bad idea, haha. h4 probably is the way to go.


**Another note...**

- **Gap on trigger:** I previously hadn't tested with long text in the trigger, but we should add a gap (like `.75em`) between the icon and tigger text:

<img width="681" alt="image" src="https://github.com/user-attachments/assets/c1e82b10-34a5-40f0-875b-976e90e29cf6">

Nyiriland commented:
Another Q for @jffng... following up on the chat we had yesterday, I see there _are_ focus styles applied at the block level, specifically to this class:

<img width="354" alt="image" src="https://github.com/user-attachments/assets/7012265f-90e7-49d1-8264-c369db00fcb4">

I previously thought these were inherited, but if they're not, can we remove it from the `:focus` and just keep on `focus:visible`, or does that need to stay to match button focus states? (I threw in a button on that [test page](https://cool-tools.mystagingwebsite.com/test/) and it actually doesn't match those, either.)

jffng commented:
> Can we add ability a border to the accordion group? 

👍 

> I'd vote for None as text

👍 

> `Automatically close this accordion when visitors open another one.`

I find it a little confusing what accordion is being referred to by "this" and "another one" — is it the accordion group or accordion item? Technically this is a setting of the group, but it affects the behavior of accordions inside the group. Also something to consider is when this setting is enabled and multiple accordion items are open by default, clicking any open section actually closes all of them. 

What do you think about `Clicking one accordion section automatically closes all other open sections.`? 

> is there any guidance around semantic hierarchy of DOM elements for block defaults? 

I think it depends what's on the page already and where the accordion appears (i.e. is there a h2 placed right before it that acts to describe / label the accordion?). Then h3 would be the correct level. 

Also different themes and style variations are going to change the size of an h3/4/5. With that in mind, do you still think h4 or h5 is best for the default, or can we keep it h3?

> can we remove it from the :focus and just keep on focus:visible

👍 

FYI I'm going to continue working on this over here, can we pick up the discussion / feedback there? 🙏 

https://github.com/WordPress/gutenberg/pull/64119/
